stor/dx.py

```diff
@@ -96,4 +91,66 @@ def exists(self):
 
         For directories on swift, checks whether directory sentinel exists or
         at least one subdirectory exists"""
-        raise NotImplementedError
\ No newline at end of file
+        raise NotImplementedError
+
+
+class DXVirtualPath(DXPath):
+
+    @cached_property
+    def canonical_project(self):
+        """Returns the first project that matches the name that user has view access to.
+        If no match is found, returns None
+        """
+        # we need the DX project-ID to get the file_ID and full path
+        proj_dict = db.search.find_one_project(
+            name=self.project, level='VIEW', zero_ok=True, more_ok=False)
```

Comment:
1. name dxb if possible
2. test case for if two+ projects with doc on error message generated from that